ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js

```diff
@@ -110,6 +110,38 @@ export function catalogSelectorSearchFilter() {
     }
 }
 
+export function catalogSelectorFiltersFilter() {
+    // compute counts and apply active filters;     
+    // this is called by the view after filtering based on search,
+    // so filters can adjust based on number of search results
+    return function (items, $scope) {
```

Comment:
Passing scope around is a bad idea for performances, you should pass each individual parameter, or an object containing all them instead.

Reply:
we set variables on the scope when the filter runs.  i take your point a nested object would be cleaner but let's do performance analysis before optimizing.  this doesn't seem to be that inefficient.

Reply:
https://stackoverflow.com/questions/17596246/access-scope-variables-from-a-filter-in-angularjs gives good caveats on this